### DISCUSSION TITLE
How do I get an access to an instance of extrastore?

### DISCUSSION THREAD

**Discussion by LeeWeeder** [patrykandpatrick/vico#785] [Questions] [answered]
### Question

Hello wonderful persons!

So, I want to know how to get an instance of extrastore. As on some functions, an extrastore instance is already exposed through a callback such as for `rememberHorizontalLine()`'s `y` parameter. I would like to use the data stored in extrastore as the value for the persistentMarkers. Although I could just take it as a parameter, I would like to maintain the function that uses it to just only take a model producer and a function that will load the data. 

My question is, how can I get an access to an instance of the extrastore for that modelProducer?

Thank you!

### Vico version(s)

2.0.0-alpha.21

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @LeeWeeder. `ExtraStore` access must occur via the parameters of lambdas, interface functions, etc. This relates to synchronization and parallel processing.

`persistentMarkers` was converted into an `ExtraStore` lambda in Vico `2.0.0-alpha.23`, making it possible to do what you've described.
